parlai/tasks/dialogue_safety/agents.py

```diff
@@ -0,0 +1,533 @@
+            build_data.download(url, dpath, fname)
+
+        # Mark the data as built.
+        build_data.mark_done(dpath, version)
+
+
+class StandardTeacher(FixedDialogTeacher):
+    """
+    Data from the standard collection described in the paper
+    `Build it Break it Fix it for Dialogue Safety: Robustness from
+    Adversarial Human Attack` (<https://arxiv.org/abs/1908.06083>)
+
+    To see data from rounds 1, 2, and 3, try running:
+    `python examples/display_data.py -t dialogue_safety:standard --round 3`
+
+    To see data from round 2 only, try running:
+    `python examples/display_data.py -t dialogue_safety:standard --round 2
+     --round-only True`
+    """
+
+    @staticmethod
+    def add_cmdline_args(parser):
+        parser = parser.add_argument_group('Standard Safety Teacher Args')
+        parser.add_argument(
+            '--round',
+            type=int,
+            default=1,
+            choices=[1, 2, 3],
+            help='Which round of data to use',
+        )
+        parser.add_argument(
+            '--round-only',
+            type='bool',
+            default=False,
+            help='if False, includes all rounds up to including the specified '
+            'round; if True, only includes data from the specified round',
+        )
+        return parser
+
+    def __init__(self, opt, shared=None):
+        build(opt['datapath'])  # download the data
+        self.opt = opt
+        self.data_path = os.path.join(
+            opt['datapath'], 'dialogue_safety', SINGLE_TURN_DATA
+        )
+
+        self.fixed_random = random.Random(42)
+        self.round = opt['round']
+        self.label_candidates = [NOT_OK_CLASS, OK_CLASS]
+
+        if shared and 'data' in shared:
+            self.data = shared['data']
+        else:
+            self._setup_data(opt['datatype'])
+
+        opt = copy.deepcopy(opt)
+        super().__init__(opt, shared)
+        self.reset()
+
+    def num_episodes(self):
+        return len(self.data)
+
+    def num_examples(self):
+        return len(self.data)
+
+    def _setup_data(self, datatype):
+        # load data
+        self.data_dump = json.load(open(self.data_path, 'rb'))['standard']
+        d_type = datatype.split(':')[0]
+        self.data = []
+        if not self.opt['round_only']:
+            # loop and add other rounds
+            for i in range(self.round - 1):
+                rnd = str(i + 1)
+                for x in ['good', 'bad']:
+                    self.data += self.data_dump[d_type][rnd][x]
+
+        # add data from current round
+        for x in ['good', 'bad']:
+            self.data += self.data_dump[d_type][str(self.round)][x]
+
+        self.fixed_random.shuffle(self.data)
+
+    def get(self, episode_idx, entry_idx):
+        return Message(self.data[episode_idx])
+
+    def share(self):
+        shared = super().share()
+        shared['data'] = self.data
+        return shared
+
+
+class AdversarialTeacher(FixedDialogTeacher):
```

Comment:
this looks identical to the `StandardTeacher` except for one line in `_setup_data`, perhaps this can subclass `Standard` and just change that one line?